samanthas_telegram_bot/api_clients/smalltalk/smalltalk_client.py

```diff
@@ -0,0 +1,273 @@
+        """Gets results of SmallTalk interview."""
+
+        user_data = context.user_data
+
+        while True:
+            logger.info("Trying to receive results from SmallTalk")
+
+            try:
+                result = await cls._process_data(
+                    update=update,
+                    context=context,
+                    data=await cls._get_data(
+                        update=update, context=context, test_id=user_data.student_smalltalk_test_id
+                    ),
+                )
+            except SmallTalkClientError as err:
+                user_data.comment = (
+                    f"{user_data.comment}\n- Could not load results of SmallTalk assessment\n"
+                    f"Interview ID: {user_data.student_smalltalk_test_id}"
+                )
+                raise SmallTalkClientError(
+                    "Could not process SmallTalk results for "
+                    f"{user_data.first_name} {user_data.last_name}"
+                ) from err
+
+            attempts = 0
+
+            if result.status == SmalltalkTestStatus.NOT_STARTED_OR_IN_PROGRESS:
+                await logs(
+                    bot=context.bot,
+                    update=update,
+                    text=(
+                        f"User {user_data.first_name} {user_data.last_name} "
+                        f"didn't finish the SmallTalk assessment."
+                    ),
+                    needs_to_notify_admin_group=True,
+                )
+                user_data.comment = (
+                    f"{user_data.comment}\n- SmallTalk assessment not finished\nCheck {result.url}"
+                )
+                return None
+            elif result.status == SmalltalkTestStatus.RESULTS_NOT_READY:
+                if attempts > MAX_ATTEMPTS_TO_GET_DATA_FROM_API:
+                    total_seconds_waiting = (
+                        MAX_ATTEMPTS_TO_GET_DATA_FROM_API
+                        * SMALLTALK_TIMEOUT_IN_SECS_BETWEEN_API_REQUEST_ATTEMPTS
+                    )
+                    await logs(
+                        bot=context.bot,
+                        update=update,
+                        level=LoggingLevel.ERROR,
+                        text=(
+                            f"SmallTalk results for {user_data.first_name} "
+                            f"{user_data.last_name} still not ready after "
+                            f"{total_seconds_waiting / 60} minutes. "
+                            f"Interview ID {user_data.student_smalltalk_test_id}."
+                        ),
+                        needs_to_notify_admin_group=True,
+                    )
+                    user_data.comment = (
+                        f"{user_data.comment}\n- SmallTalk assessment results were not ready\n"
+                        f"Interview ID {user_data.student_smalltalk_test_id}"
+                    )
+
+                logger.info("SmallTalk results not ready. Waiting...")
+                attempts += 1
```

Comment:
You have method with retries in base client, why you don't use it here?

Reply:
Maybe add something like `RetryRule` parameter to this method, with for example statuses which have to be retried.

Reply:
I'll answer in a DM :)